### PR TITLE
DOC: fix bugs in signal.TransferFunction docstring.  Closes gh-5287.

### PR DESCRIPTION
**Pull request by rgommers**

### CODE REVIEW

File: scipy/signal/ltisys.py [hunk 1]
```diff
@@ -578,8 +578,8 @@ class TransferFunction(lti):
     """Linear Time Invariant system class in transfer function form.
 
     Represents the system as the transfer function
-    :math:`H(s)=\sum_i b[i] s^i / \sum_j a[j] s^i`, where :math:`a` are
-    elements of the numerator `num` and :math:`b` are the elements of the
+    :math:`H(s)=\sum_{i=0}^N b[N-i] s^i / \sum_{j=0}^M a[M-j] s^i`, where :math:`b` are
```
Comment (WarrenWeckesser) [inline scipy/signal/ltisys.py]: The term in the summation in the denominator should be `s^j`, or the dummy variable in the should be changed to `i`.

Comment (rgommers) [inline scipy/signal/ltisys.py]: Good catch, fixed. That was bug number 3 in that one paragraph.

**Comment by pv** [comment]
LGTM

File: scipy/signal/ltisys.py [hunk 2]
```diff
@@ -578,8 +578,8 @@ class TransferFunction(lti):
     """Linear Time Invariant system class in transfer function form.
 
     Represents the system as the transfer function
-    :math:`H(s)=\sum_i b[i] s^i / \sum_j a[j] s^i`, where :math:`a` are
-    elements of the numerator `num` and :math:`b` are the elements of the
+    :math:`H(s)=\sum_{i=0}^N b[N-i] s^i / \sum_{j=0}^M a[M-j] s^j`, where :math:`b` are
```
Comment (matthew-brett) [inline scipy/signal/ltisys.py]: Forgive my ignorance, but is `N == len(b) - 1`?

Comment (rgommers) [inline scipy/signal/ltisys.py]: indeed, I'll send a fix to clarify that.

**Pull request merged**